exporter/otlphttpexporter/otlp.go

```diff
@@ -223,25 +223,46 @@ func (e *baseExporter) export(ctx context.Context, url string, request []byte, p
 	formattedErr = httphelper.NewStatusFromMsgAndHTTPCode(errString, resp.StatusCode).Err()
 
 	if isRetryableStatusCode(resp.StatusCode) {
-		// A retry duration of 0 seconds will trigger the default backoff policy
-		// of our caller (retry handler).
-		retryAfter := 0
-
 		// Check if the server is overwhelmed.
 		// See spec https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#otlphttp-throttling
 		isThrottleError := resp.StatusCode == http.StatusTooManyRequests || resp.StatusCode == http.StatusServiceUnavailable
-		if val := resp.Header.Get(headerRetryAfter); isThrottleError && val != "" {
-			if seconds, err2 := strconv.Atoi(val); err2 == nil {
-				retryAfter = seconds
+		if isThrottleError {
+			// A retry duration of 0 seconds will trigger the default backoff policy of our caller (retry handler).
+			throttleDuration := getRetryAfterSeconds(resp)
+			if throttleDuration != 0 {
+				return exporterhelper.NewThrottleRetry(formattedErr, throttleDuration)
 			}
+			return formattedErr
```

Comment:
We still need to return `exporterhelper.NewThrottleRetry` when the header isn't set and falls back to the default intervals

Reply:
Why?

Reply:
I think I agree -- if the server returned a RetryAfter, even a 0 or an already-past timestamp, maintaining the value makes sense.

Reply:
> even a 0 or an already-past timestamp, maintaining the value makes sense.

What if is invalid?

Reply:
Changed to preserve if the header is present and has a valid value (based on the RFC).

Reply:
Currently, we always return `throttleRetry` if `isRetryableStatusCode(resp.StatusCode)` (true for 4 codes). This PR changes the behavior to return `throttleRetry` only for `http.StatusTooManyRequests` and `http.StatusServiceUnavailable`. Is that an expected change?

Reply:
Nevermind. I see that `throttleRetry` isn't needed for the retry sender to retry